### PR TITLE
User (devise) was created.

### PR DESCRIPTION
**Pull request by maruraru**
Now admin can create users. 
http basic auth is used.

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,3 +1,9 @@
 class ApplicationController < ActionController::Base
 	
+	def authenticate_admin_user
```
Comment (AlexeyBoldak) [inline app/controllers/application_controller.rb]: indentation
You can use rubocop for fixing it

File: config/application.rb [hunk 2]
```diff
@@ -6,6 +6,8 @@
 # you've limited to :test, :development, or :production.
 Bundler.require(*Rails.groups)
 
+
+
```
Comment (AlexeyBoldak) [inline config/application.rb]: Remove empty lines

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (maruraru) [reply app/controllers/application_controller.rb]: Done.
Tomorrow I'll start using RubyMine with rubocop

File: config/application.rb (hunk 2 above)
Comment (maruraru) [reply config/application.rb]: Removed.

File: .env [hunk 3]
```diff
@@ -0,0 +1,2 @@
+ADMIN_NAME=admin
```
[the remark this answers is not in the document]
Comment (maruraru) [reply .env]: done

**Comment by AlexeyBoldak** [comment]
1. Add validations for User
2. Add tests for validation
3. Use Faker for FactoryBot
4. Fix schema.rb

File: config/routes.rb [hunk 4]
```diff
@@ -1,7 +1,7 @@
 Rails.application.routes.draw do
-
+  get "application/main_page"
+  root "application#main_page"
```
[the remark this answers is not in the document]
Comment (maruraru) [reply config/routes.rb]: done

File: spec/factories.rb [hunk 5]
```diff
@@ -0,0 +1,8 @@
+FactoryBot.define do
+    factory :user do
+      organisation_name { 'test4' }
```
[the remark this answers is not in the document]
Comment (maruraru) [reply spec/factories.rb]: done

File: spec/features/user_authorization_spec.rb [hunk 6]
```diff
@@ -0,0 +1,44 @@
+require 'rails_helper'
+require 'database_cleaner'
+
+DatabaseCleaner.strategy = :truncation
+DatabaseCleaner.start
```
[the remark this answers is not in the document]
Comment (maruraru) [reply spec/features/user_authorization_spec.rb]: done

File: spec/features/user_registration_spec.rb [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+    end
+  end
+
+  let(:user2) { FactoryBot.create(:user) }
+
+  context 'register fails' do
+    it 'has not unique email' do
+      visit '/users/sign_up'
+      within("#new_user") do
+        fill_in 'user_email', :with => user2.email
```
[the remark this answers is not in the document]
Comment (maruraru) [reply spec/features/user_registration_spec.rb]: done

File: app/models/user.rb [hunk 8]
```diff
@@ -1,3 +1,5 @@
+# frozen_string_literal: true
+
```
[the remark this answers is not in the document]
Comment (maruraru) [reply app/models/user.rb]: done

File: spec/features/user_registration_spec.rb [hunk 9]
```diff
@@ -0,0 +1,77 @@
+require 'rails_helper'
+require 'database_cleaner'
+
+DatabaseCleaner.strategy = :truncation
+DatabaseCleaner.start
```
[the remark this answers is not in the document]
Comment (maruraru) [reply spec/features/user_registration_spec.rb]: +

File: spec/features/user_registration_spec.rb [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+        fill_in 'user_password', with: user2.password
+        fill_in 'user_password_confirmation', with: ''
+      end
+      click_button 'Sign up'
+      expect(page).to have_text 'can\'t be blank'
+    end
+  end
+end
+
+DatabaseCleaner.clean
```
[the remark this answers is not in the document]
Comment (maruraru) [reply spec/features/user_registration_spec.rb]: +

File: spec/models/user_spec.rb [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+      user.email = user_data.email
+      user.organisation_name = user_data.organisation_name
+      user.password = user_data.password
+      user.password_confirmation = user_data.password_confirmation
+      expect(user).to be_valid
+    end
+  end
+
+  context 'some fields sre incorrect' do
+    it 'isn\'t valid w/o org name' do
```
[the remark this answers is not in the document]
Comment (maruraru) [reply spec/models/user_spec.rb]: done

File: spec/factories.rb [hunk 12]
```diff
@@ -0,0 +1,11 @@
+require 'faker'
```
[the remark this answers is not in the document]
Comment (maruraru) [reply spec/factories.rb]: done

**Pull request merged**